src/write_gcm.c

```diff
@@ -597,26 +596,27 @@ static int wg_curl_get_or_post(char *response_buffer,
   return result;
 }
 
-static size_t wg_curl_write_callback(char *ptr, size_t size, size_t nmemb,
+static size_t wg_curl_write_callback(void *ptr, size_t size, size_t nmemb,
                                      void *userdata) {
-  wg_curl_write_ctx_t *ctx = userdata;
-  if (ctx->size == 0) {
+  size_t requested_bytes = size * nmemb;
+  wg_curl_write_ctx_t *ctx = (wg_curl_write_ctx_t *) userdata;
+
+  // TODO: This is a potential performance bottleneck. We should consider
+  //       doubling the buffer size to minimize the number of copies.
+  char *new_data = realloc(ctx->data, ctx->size + requested_bytes + 1);
+  if (new_data == NULL) {
+    ERROR("wg_curl_write_callback: not enough memory, tried to allocate %zu "
+          "bytes (realloc returned NULL)", ctx->size + requested_bytes + 1);
```

Comment:
Same comment here -- better to have the spaces in strings at the start of the line rather than at the end.